chips/sam4l/src/acifc.rs

```diff
@@ -0,0 +1,370 @@
+		ACTEST7 7,
+		ACTEST6 6,
+		ACTEST5 5,
+		ACTEST4 4,
+		ACTEST3 3,
+		ACTEST2 2,
+		ACTEST1 1,
+		ACTEST0 0
+	],
+
+	Parameter [
+		// If equal to one, window mode x is implemented.
+		WIMPL3 19,
+		WIMPL2 18,
+		WIMPL1 17,
+		WIMPL0 16,
+		// If equal to one, analog comparator x is implemented.
+		ACIMPL7 7,
+		ACIMPL6 6,
+		ACIMPL5 5,
+		ACIMPL4 4,
+		ACIMPL3 3,
+		ACIMPL2 2,
+		ACIMPL1 1,
+		ACIMPL0 0
+		],
+
+	WindowConfiguration [
+		// If equal to one, window mode is enabled.
+		WFEN OFFSET(16) NUMBITS(1) [],
+		// If equal to one, peripheral event from ACWOUT is enabled.
+		WEVEN OFFSET(11) NUMBITS(1) [],
+		// Peripheral Event Source Selection for Window Mode
+		WEVSRC OFFSET (8) NUMBITS(3) [
+			AcwoutRisingEdge = 0,
+			AcwoutFallingEdge = 1,
+			AcwoutRisingOrFallingEdge = 2,
+			InsideWindow = 3,
+			OutsideWindow = 4,
+			MeasureDone = 5
+		],
+			// Window Mode Interrupt Settings
+		WIS OFFSET(0) NUMBITS (3)[
+			// Window interrupt as soon as the common input voltage is inside the window
+			InterruptInsideWindow = 0,
+			// Window interrupt as soon as the common input voltage is outside the window
+			InterruptOutsideWindow = 1,
+			// Window interrupt on toggle of ACWOUT
+			InterruptToggleAcwout = 2,
+			// Window interrupt when evaluation of common input voltage is done
+			InterruptAfterEvaluation = 3,
+			// Window interrupt when the common input voltage enters the window (i.e., rising-edge of ACWOUT)
+			InterruptEnterWindow = 4,
+			// Window interrupt when the common input voltage leaves the window (i.e., falling-edge of ACWOUT)
+			InterruptLeaveWindow = 5
+	]
+	],
+
+	ACConfiguration [
+		// If equal to one, AC is always enabled. 
+		ALWAYSON OFFSET(27) NUMBITS(1) [],
+		// 0: Low-power mode. 1: Fastm ode.
+		FAST OFFSET(26) NUMBITS(1) [],
+		// Hysteresis voltage value: 0/25/50/75 mV
+		HYS OFFSET(24) NUMBITS(2) [
+			HysteresisVoltage0mV = 0,
+			HysteresisVoltage25mV = 1,
+			HysteresisVoltage50mV = 2,
+			HysteresisVoltage75mV = 3
+		],
+		// Setting this to one will output peripheral event when ACOUT is zero.
+		EVENP OFFSET(17) NUMBITS(1) [],
+		// Setting this to one will output peripheral event when ACOUT is one.
+		EVENN OFFSET(16) NUMBITS(1) [],
+		// Negative input select. 00: ACANx pint selected, others reserved.
+		INSELN OFFSET(8) NUMBITS(2) [],
+		// Choose between analog comparator mode.
+		MODE OFFSET(4) NUMBITS(2) [
+			Off = 0,
+			ContinuousMeasurementMode = 1,
+			// User Triggered Single Measurement Mode
+			UserMode = 2,
+			// Peripheral Event Single Measurement Mode
+			PeripheralMode = 3
+		],
+		// Interrupt settings
+		IS OFFSET(0) NUMBITS(2) [
+			// When Vinp > Vinn
+			WhenVinpGtVinn = 0,
+			// When Vinp < Vinn
+			WhenVinpLtVinn = 1,
+			// On toggle of ACOUT
+			OnToggleOfACOUT = 2,
+			// When comparison of Vinp and Vinn is done
+			WhenComparisonDone = 3
+		]
+	]
+];
+
+const BASE_ADDRESS: *mut AcifcRegisters = 0x40040000 as *mut AcifcRegisters;
```

Comment:
Add a line here like:
`const ACIFC_REGS: StaticRef<AcifcRegisters> = unsafe { StaticRef::new(BASE_ADDRESS as *const AcifcRegisters) };`

Then everywhere below you can just use ACICF_REGS without unsafe